### PR TITLE
Cleaning up emily's user-header implementation

### PR DESCRIPTION
**Pull request by sanghun89**
## Description
- Implemented user header that emily wrote with cleaner git history.
- Fixed up issues when embedding to user's actual `show.html.erb`
- Only implemented designs - no data attached
- No responsive for this (Did not want to put more than what it already had)

## Trello Link
https://trello.com/c/W3UjhGHJ/42-logged-in-homedesign-implementation-as-a-user-i-want-to-see-my-basic-profile-page

## Screen Shots
Screen Shot 2019-03-17 at 1 38 24 AM

### CODE REVIEW

**Comment by markdamobi** [comment]
@sanghun89 @emilyspieler not a huge deal, but can the name of the branch be changed? The user home page is a different task and it's not being implemented on this branch.

**Comment by sanghun89** [comment]
> @emilyspieler @sanghun89 Looks like the link attached to the description is for the header in the home page. I don't see a trello card specifically for the user header on the profile page. This branch doesn't include the "Edit Profile" button that's showing [on zeplin](https://app.zeplin.io/project/5c5390a362086d356556a647/screen/5c669bcfa6682599c7256eaa). Shouldn't that be included in this branch?

I wanted to fix up what was on there as-is, but it seems like that is something missing from @emilyspieler 's original work. Could you implement that on this new branch ??

**Comment by sanghun89** [comment]
> @sanghun89 @emilyspieler not a huge deal, but can the name of the branch be changed? The user home page is a different task and it's not being implemented on this branch.

This was the original branch name - I think that can be ignored having the PR title actually reflecting what is being changed

**Comment by markdamobi** [comment]
> This was the original branch name - I think that can be ignored having the PR title and description actually reflecting what is being changed, which I opinion is more important as those need to actually state what you have done to inform other developers the context around the PR.

That's fine. This one can be ignored, but it's something to keep in mind for future prs. I pointed it out because the original branch had some work done with creating/modifying controllers, but I don't see any of that work in this branch. Moving forward, it wouldn't be a bad idea to have the pr title be related to the branch name.

**Comment by sanghun89** [comment]
> > This was the original branch name - I think that can be ignored having the PR title and description actually reflecting what is being changed, which I opinion is more important as those need to actually state what you have done to inform other developers the context around the PR.
> 
> That's fine. This one can be ignored, but it's something to keep in mind for future prs. I pointed it out because the original branch had some work done with creating/modifying controllers, but I don't see any of that work in this branch. Moving forward, it wouldn't be a bad idea to have the pr title be related to the branch name.

It was originally named that thinking that was the page the story was to implement, but through last two plannings/retro we had decided to remove it and only apply the layout. I've only named it very identical to indicate which branch that I am "fix"ing

That being said - yes it would very much to be wise to name your branch to the work that is actually being implemented, but it is negligible if one had enough of detail written in the PR title and description.

**Comment by mintii** [comment]
All sounds good to me @sanghun89 👍

File: app/assets/stylesheets/layouts/user_header.scss [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      }
+
+      &.active, &:hover {
+        text-decoration: none;
+        background: $theme-secondary;
+        color: $text-light;
+      }
+    }
+  }
+}
```
Comment (arincon9) [inline app/assets/stylesheets/layouts/user_header.scss]: Can we please add an extra line

File: app/views/layouts/user_header.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      <a href="#" class="active">
+        <%= fa_icon "user-circle" %>
+        <span>Profile</span>
+      </a>
+      <a href="#">
+        <%= fa_icon "cog" %>
+        <span>Settings</span>
+      </a>
+    </nav>
+</div>
```
Comment (arincon9) [inline app/views/layouts/user_header.html.erb]: Can we please add an extra line?

File: app/views/layouts/user_header.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<div class="user-header d-flex container-fluid">
+    <div class="user-header__image">
+      <%= image_tag("home-profile_01.jpg", :alt => "rss feed", :class=> "rounded-circle") %>
```
Comment (arincon9) [inline app/views/layouts/user_header.html.erb]: Minor: space between class so: `:class => "rounded-circle"`

**Comment by uuunk** [comment]
LGTM as well. @markdamobi @sanghun89 @emilyspieler we'll make a new card for the header on the profile page that includes edit profile. https://trello.com/c/MBNIEj1e/57-profileheader-as-a-user-i-want-to-be-able-to-have-site-level-context-and-be-able-to-edit-my-profile-on-when-on-the-profile-page

**Pull request merged**